tests.c

```diff
@@ -11901,6 +11901,53 @@ test_decode_response_header_kmip_2_0(TestTracker *tracker)
     return(result);
 }
 
+int
+test_compare_query_functions(TestTracker *tracker)
+{
+    TRACK_TEST(tracker);
+
+    uint8 encoding[32] = {
+        0x42, 0x00, 0x74, 0x05, 0x00, 0x00, 0x00, 0x04,
+        0x00, 0x00, 0x00, 0x01, 0x00, 0x00, 0x00, 0x00,
+        0x42, 0x00, 0x74, 0x05, 0x00, 0x00, 0x00, 0x04,
+        0x00, 0x00, 0x00, 0x02, 0x00, 0x00, 0x00, 0x00
+    };
+
+    KMIP ctx = {0};
+    kmip_init(&ctx, encoding, ARRAY_LENGTH(encoding), KMIP_1_0);
+
+    LinkedList list_1 = {0};
+    LinkedListItem item_1 = {0};
+    int32 funct_1 = KMIP_QUERY_OPERATIONS;
+    item_1.data = &funct_1;
+
+    LinkedListItem item_2 = {0};
+    int32 funct_2 = KMIP_QUERY_SERVER_INFORMATION;
+    item_2.data = &funct_2;
+
+    kmip_linked_list_enqueue(&list_1, &item_1);
+    kmip_linked_list_enqueue(&list_1, &item_2);
+
+    Functions expected = {0};
+    expected.function_list = &list_1;
+
+    Functions observed = {0};
+
+    int result = kmip_decode_query_functions(&ctx, &observed);
+    int comparison = kmip_compare_query_functions(&expected, &observed);
+    if(!comparison)
+    {
+        // comparision is expected to fail
```

Comment:
Why is this expected to fail? If you want to test a failed comparison, you'll need to add an `else` case that reports a failed test if the expected query functions match the observed ones.

Reply:
I changed the test to do a normal comparison